cpp/src/arrow/flight/client.h

```diff
@@ -108,10 +108,14 @@ class ARROW_FLIGHT_EXPORT FlightClientOptions {
   /// positive. When enabled, FlightStreamWriter.Write* may yield a
   /// IOError with error detail FlightWriteSizeStatusDetail.
   int64_t write_size_limit_bytes;
+
   /// \brief Generic connection options, passed to the underlying
   ///     transport; interpretation is implementation-dependent.
   std::vector<std::pair<std::string, util::variant<int, std::string>>> generic_options;
 
+  /// \brief Use TLS without validating the server certificate. Use with caution.
+  bool disable_server_verification;
+
   /// \brief Get default options.
   static FlightClientOptions Defaults();
```

Comment:
I believe you're seeing failures since `Defaults` should be updated to set the new flag, else the flag may or may not be initialized.

Reply:
I've added an explicitly defined constructor since users might be constructing FlightCredentialsOptions directly. I am surprised that this didn't previously cause problems with the write_size_limit_bytes option though.

Reply:
The intent was to use Defaults(), at least.